peer/brontide.go

```diff
@@ -1741,65 +1741,37 @@ func (p *Brontide) storeError(err error) {
 	)
 }
 
-// handleWarning processes a warning message read from the remote peer. The
-// boolean return indicates whether the message should be delivered to a
-// targeted peer or not. The message gets stored in memory as an error if we
-// have open channels with the peer we received it from.
+// handleWarningOrError processes a warning or error msg and returns true if
+// msg should be forwarded to the associated channel link. False is returned if
+// any necessary forwarding of msg was already handled by this method. If msg is
+// an error from a peer with an active channel, we'll store it in memory.
 //
 // NOTE: This method should only be called from within the readHandler.
-func (p *Brontide) handleWarning(msg *lnwire.Warning) bool {
-	switch {
-	// Connection wide messages should be forward the warning to all the
-	// channels with this peer.
-	case msg.ChanID == lnwire.ConnectionWideID:
-		for _, chanStream := range p.activeMsgStreams {
-			chanStream.AddMsg(msg)
-		}
-
-		return false
-
-	// If the channel ID for the warning message corresponds to a pending
-	// channel, then the funding manager will handle the warning.
-	case p.cfg.FundingManager.IsPendingChannel(msg.ChanID, p):
-		p.cfg.FundingManager.ProcessFundingMsg(msg, p)
-		return false
+func (p *Brontide) handleWarningOrError(chanID lnwire.ChannelID,
+	msg lnwire.Message) bool {
 
-	// If not we hand the warning to the channel link for this channel.
-	case p.isActiveChannel(msg.ChanID):
-		return true
-
-	default:
-		return false
+	if errMsg, ok := msg.(*lnwire.Error); ok {
+		p.storeError(errMsg)
 	}
-}
-
-// handleError processes an error message read from the remote peer. The boolean
-// returns indicates whether the message should be delivered to a targeted peer.
-// It stores the error we received from the peer in memory if we have a channel
-// open with the peer.
-//
-// NOTE: This method should only be called from within the readHandler.
-func (p *Brontide) handleError(msg *lnwire.Error) bool {
-	// Store the error we have received.
-	p.storeError(msg)
 
 	switch {
-	// In the case of an all-zero channel ID we want to forward the error to
```

Comment:
An alternative is to keep `handleWarning` as-is, and call `p.handleWarning` inside `handleError`, which can also remove some duplications. But a more important question is do we handle them differently from specs level?

Reply:
> An alternative is to keep `handleWarning` as-is, and call `p.handleWarning` inside `handleError`, which can also remove some duplications.

Yes, but then `handleWarning` isn't actually handling warnings.

> But a more important question is do we handle them differently from specs level?

I don't understand the question.  This PR doesn't change functionality at all.

Reply:
> Yes, but then handleWarning isn't actually handling warnings.

yeah that's also true.

> I don't understand the question. This PR doesn't change functionality at all.

my bad. Was wondering if we needed separate methods in case we wanna handle them differently in `brontide`. And was wondering how the specs differentiate handling these two.